src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/persist/SQLParsingUtil.java

```diff
@@ -70,18 +71,16 @@ private static Record getRecordByName(Result<Record> result, Field<String> field
    */
   public static double readDataFromSqlResult(Result<Record> result, Field<String> matchedField, String matchedFieldName, String dataField) {
     double ret = Double.NaN;
-    if (result != null) {
-      try {
-        Record record = SQLParsingUtil.getRecordByName(result, matchedField, matchedFieldName);
-        ret = record.getValue(MetricsDB.MAX, Double.class);
-      }
-      catch (IllegalArgumentException ie) {
-        LOG.error("{} fails to match any row in field {}.", matchedFieldName, matchedField.getName());
-      }
-      catch (
-          DataTypeException de) {
-        LOG.error("Fail to read {} field from SQL result", dataField);
-      }
+    try {
+      Record record = getRecordByName(result, matchedField, matchedFieldName);
+      ret = record.getValue(MetricsDB.MAX, Double.class);
+    }
+    catch (IllegalArgumentException ie) {
```

Comment:
Can we add a test to check Illegal Argument Exception and DataTypeException is thrown ?

Reply:
Discussed how this is covered